compiler/src/dotty/tools/dotc/transform/ElimRepeated.scala

```diff
@@ -109,35 +112,41 @@ class ElimRepeated extends MiniPhase with InfoTransformer { thisPhase =>
   override def transformTypeApply(tree: TypeApply)(implicit ctx: Context): Tree =
     transformTypeOfTree(tree)
 
-  /** If method overrides a Java varargs method, add a varargs bridge.
-   *  Also transform trees inside method annotation
+  /** If method overrides a Java varargs method or is annotated with @varargs, add a varargs bridge.
+   *  Also transform trees inside method annotation.
    */
   override def transformDefDef(tree: DefDef)(implicit ctx: Context): Tree =
     ctx.atPhase(thisPhase) {
-      if (tree.symbol.info.isVarArgsMethod && overridesJava(tree.symbol))
-        addVarArgsBridge(tree)
+      val isOverride = overridesJava(tree.symbol)
+      val hasAnnotation = hasVargsAnnotation(tree.symbol)
+      if tree.symbol.info.isVarArgsMethod && (isOverride || hasAnnotation) then
+        // java varargs can be overriden by synthetic methods with an Array param,
+        // but non-overrides need the varargs bytecode flag and cannot be synthetic
```

Comment:
For simplicity, I would always emit the varargs flag even for the synthetic overrides.